unittest/compare_ints.c

```diff
@@ -126,7 +127,7 @@ int main(int argc, char *argv[]) {
         uint64_t err = llabs(data1 - data2);
         if (err > tol) {
             errors ++;
-            printf("%d %lld %lld\n", count, data1, data2);
+            printf("%d %" PRId64 " %" PRId64 "\n", count, data1, data2);
```

Comment:
Seems weird that gcc10 doesn't support %lld, no?

Reply:
Actually I'm seeing a warning for this when I build on gcc 9.3.0:
```
/home/david/codec2/unittest/compare_ints.c:129:27: warning: format ‘%lld’ expects argument of type ‘long long int’, but argument 3 has type ‘int64_t’ {aka ‘long int’} [-Wformat=]
  129 |             printf("%d %lld %lld\n", count, data1, data2);

```

Reply:
So it's not lack of support for %lld, just the wrong data type

Reply:
> So it's not lack of support for %lld, just the wrong data type

That depends on the arch (on x64 `int64_t` normally maps to long, on i386 to long long). That's why you should use the macros to make it portable.